Userland/Libraries/LibSemVer/SemVer.cpp

```diff
@@ -0,0 +1,290 @@
+/*
+ * Copyright (c) 2023, Gurkirat Singh <tbhaxor@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/CharacterTypes.h>
+#include <AK/Find.h>
+#include <AK/GenericLexer.h>
+#include <AK/ReverseIterator.h>
+#include <AK/StringBuilder.h>
+#include <LibSemVer/SemVer.h>
+
+namespace SemVer {
+String SemVer::suffix() const
+{
+    StringBuilder sb;
+    if (!m_prerelease_identifiers.is_empty())
+        sb.appendff("-{}", prerelease());
+    if (!m_buildmetadata.is_empty())
+        sb.appendff("+{}", buildmetadata());
+    return sb.to_string().release_value_but_fixme_should_propagate_errors();
+}
+
+String SemVer::to_string() const
+{
+    return String::formatted("{}{}{}{}{}{}", m_major, m_number_separator, m_minor, m_number_separator, m_patch, suffix()).release_value_but_fixme_should_propagate_errors();
+}
+
+SemVer SemVer::bump(BumpType type) const
+{
+    switch (type) {
+    case BumpType::Major:
+        return SemVer(m_major + 1, 0, 0, m_number_separator);
+    case BumpType::Minor:
+        return SemVer(m_major, m_minor + 1, 0, m_number_separator);
+    case BumpType::Patch:
+        return SemVer(m_major, m_minor, m_patch + 1, m_number_separator);
+    case BumpType::Prerelease: {
```

Comment:
personally I don't think that bumping pre-release can be done in a generic way, as prerelease is literally just `alphanumeric.alphanumeric.alphanumeric. ...` for as many prerelease identifiers as you like. But w/e, we can keep this.